From b360457ad692165effecfd0e3f2756f5c032e39c Mon Sep 17 00:00:00 2001
From: Rishik Ramena <rishikramena@microsoft.com>
Date: Sun, 10 Mar 2024 05:37:01 +0000
Subject: [PATCH] Removed DCHECK from BackgroundModeManager::OnProfileWillBeRemoved

This DCHECK for profile will be hit for valid scenarios where ephemeral
OTR profiles are used. In case of crashes the ephemeral profiles are not deleted on shutdown and there is logic to retry deleting them on the next browser start. So this code iterates through all the profiles and deletes the ephemeral ones. By this stage, profiles are not loaded and hence are not present in the profile manager. Hence this DCHECK for profile fails for this valid scenario.

Change-Id: Ied451f916e23772336094e344ba24fd32a3c7c15
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5331390
Reviewed-by: Alex Ilin <alexilin@chromium.org>
Reviewed-by: Nicolas Ouellet-Payeur <nicolaso@chromium.org>
Commit-Queue: Rishik Ramena <rishikramena@microsoft.com>
Cr-Commit-Position: refs/heads/main@{#1270676}
---

--- a/chrome/browser/background/background_mode_manager.cc
+++ b/chrome/browser/background/background_mode_manager.cc
@@ -553,7 +553,9 @@
     const base::FilePath& profile_path) {
   Profile* profile =
       g_browser_process->profile_manager()->GetProfileByPath(profile_path);
-  DCHECK(profile);
+  if (!profile) {
+    return;
+  }
   UnregisterProfile(profile);
 }
 
